src/bundle.c

```diff
@@ -989,34 +1010,21 @@ static enum swupd_code install_bundles(struct list *bundles, struct list **subs,
 
 		/* This was staged by verify_fix_path */
 		if (!file->staging) {
-			if (!full_mom) {
-				/* step 3: Add tracked bundles */
-				timelist_timer_start(global_times, "Add tracked bundles");
-				read_subscriptions(subs);
-				set_subscription_versions(mom, NULL, subs);
-				mom->submanifests = recurse_manifest(mom, *subs, NULL, false, NULL);
-				if (!mom->submanifests) {
-					error("Cannot load installed bundles\n");
-					ret = SWUPD_RECURSE_MANIFEST;
-					goto out;
-				}
-
-				mom->files = files_from_bundles(mom->submanifests);
-				mom->files = consolidate_files(mom->files);
-				timelist_timer_stop(global_times); // closing: Add tracked bundles
-				full_mom = mom;
-			}
-			file = search_file_in_manifest(full_mom, file->filename);
+			/* the current file struct doesn't have the name of the "staging" file
+			 * since it was staged by verify_fix_path, the staged file is in the
+			 * file struct in the MoM, so we need to load that one instead
+			 * so rename_staged_file_to_final works properly */
+			file = search_file_in_manifest(mom, file->filename);
```

Comment:
I have a great feeling that this is slowing down bundle_add a lot. Any idea why we need that on bundle_add and not on update?
Can't we use rename_all_files_to_final() instead of this loop?

Reply:
to be honest I'm not 100% sure why we need to load the file there again.... I guess the it is just so we have the correct metadata for that file since it should have been fixed by verify_fix_path

Reply:
I don't think this is that problematic because this code should only run for every file that was fixed by verify_fix_path (if any), so in theory it should not run at all or it should only run a few times.